.github/workflows/valid-files.yml

```diff
@@ -0,0 +1,55 @@
+name: Check if files are valid
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  max-file-size:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Fetch everything
+        run: git fetch --prune --unshallow
+
+      - name: Error if any file is above 1 MB
+        run: |
+          FILES="$(git diff --name-only origin/master...)"
+
+          for FILE in $FILES; do
+            if [ "$(stat --printf="%s" "$FILE")" -gt 100000 ]; then
+              echo "$FILE is larger than 10 KB"
+              echo "Please ping @iHiD if you are sure that this file is required."
+              exit 1
+            fi
+          done
+
+  valid-filenames:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Fetch everything
+        run: git fetch --prune --unshallow
+
+      - name: Error if filenames contain illegal characters or words
+        run: |
+          FILES="$(git diff --name-only origin/master...)"
+
+          for FILE in $FILES; do
+            if [[ "$FILE" =~ ^[a-zA-Z0-9._-]+$ ]]; then
```

Comment:
I would have thought the `.` and `-` need escaping here. Not sure though.

Reply:
Regex101 didn't complain. I don't know regex well enough to know for sure though

Reply:
OK, so actually it seems work fine in zsh so I guess it'll be good in bash. It'd fail in Ruby. Thanks :)

Reply:
`-` only needs to be escaped if it isn't the last character inside of `[]` since it has special meaning inside of the block.

Reply:
It also passes rubular: https://rubular.com/r/cbAS0ktyVCucNx